File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []


        for ch in s:
            if ch == '{' or ch == '(' or ch=='[':
                stack.append(ch)
            
            else:
                if ch == '}' and len(stack) != 0 and stack.pop() == '{':
                    continue
                
                elif ch == ')' and len(stack) != 0 and stack.pop() == '(':
                    continue
                
                elif ch == ']' and len(stack) != 0 and stack.pop() == '[':
                    continue

                else:
                    return False
            
        return len(stack) == 0
                

        
        

File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<int> st;

        for(auto& ch:s){
            if(ch == '(' || ch == '{' || ch == '[')
                st.push(ch);
            else{
                if(st.empty()) return false;
                char top = st.top();
                if(ch == ')' && top == '(' || ch == '}' && top=='{' || ch == ']' && top == '['){
                    st.pop();
                }
                else return false;
            }
        }
        return st.size()==0;
    }
};